pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -425,14 +426,16 @@ func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1a
 	namespaces := make([]toolchainv1alpha1.NSTemplateSetNamespace, len(nstemplateTier.Spec.Namespaces))
 	for i, ns := range nstemplateTier.Spec.Namespaces {
 		namespaces[i] = toolchainv1alpha1.NSTemplateSetNamespace{
-			Type:     ns.Type,
-			Revision: ns.Revision,
+			Type:        ns.Type,
+			Revision:    ns.Revision,
+			TemplateRef: nstemplatetiers.NewTierTemplateName(nstemplateTier.Name, ns.Type, ns.Revision),
```

Comment:
We should rather use the TemplateRef from the tier than construct the name:
```suggestion
			TemplateRef: ns.TemplateRef,
```